packages/cli/src/build/config.ts

```diff
@@ -66,7 +67,29 @@ const babelHelperId = /@babel\/runtime(|-corejs[23])\/helpers\//;
 const interop = (id: string | null): "auto" | "default" =>
   id && babelHelperId.test(id) ? "default" : "auto";
 
+function forkDevProdConditions(
+  options: OutputOptions & { entryFileNames: string; chunkFileNames: string },
+  shouldFork: boolean | undefined
+): OutputOptions[] {
+  if (!shouldFork) {
+    return [options];
+  }
+  return (["dev", "prod"] as const).map((condition) => ({
+    ...options,
+    entryFileNames: forkExtension(options.entryFileNames, condition),
+    chunkFileNames: forkExtension(options.chunkFileNames, condition),
+    plugins: [
+      inlineProcessEnvNodeEnv({
+        sourceMap: false,
+        value: condition === "prod" ? "production" : "development",
+      }),
```

Comment:
I recall we were discussing in the past using something else than `process.env.NODE_ENV` for this. It's a somewhat weird-ish convention that abuses the `process.env` in a way that it was not designed to be abused :P What are your thoughts about this? Do you want to continue using this? Or would you like to switch to some `import.meta.preconstruct.isDevelopment`?